ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterChangeHandler.java

```diff
@@ -54,4 +59,22 @@
         .map(instanceConfig -> getDataNode(instanceConfig.getInstanceName()))
         .map(dataNode -> getReplicaId(dataNode, partition.toPathString())).filter(Objects::nonNull);
   }
+
+  @Override
+  default Map<ReplicaState, List<AmbryReplica>> getSnapshotOfReplicaStates(AmbryPartition partition) {
+    String resourceName = getPartitionToResourceMap().get(partition.toPathString());
+    RoutingTableSnapshot snapshot = getRoutingTableSnapshot();
+    Map<ReplicaState, List<AmbryReplica>> replicasByState = new HashMap<>();
+    for (ReplicaState state : EnumSet.of(ReplicaState.OFFLINE, ReplicaState.BOOTSTRAP, ReplicaState.STANDBY,
```

Comment:
Just wondering why ERROR and DROPPED states aren't included here

Reply:
DROPPED replica means the replica is fully decommissioned and data files are deleted. Hence, we don't have to include it. 
ERROR replica is put into ERROR state for some reason (i.e. disk failure or data corruption etc), the data may not be trustable and we'd better exclude it.